Sources/Plasma/Apps/plFileEncrypt/CMakeLists.txt

```diff
@@ -3,12 +3,13 @@ set(plFileEncrypt_SOURCES
 )
 
 add_executable(plFileEncrypt ${plFileEncrypt_SOURCES})
-target_link_libraries(plFileEncrypt CoreLib plFile)
-target_link_libraries(plFileEncrypt ${STRING_THEORY_LIBRARIES})
-
-if(USE_VLD)
-    target_link_libraries(plFileEncrypt ${VLD_LIBRARY})
-endif()
+target_link_libraries(
+    plFileEncrypt
+    PRIVATE
+        CoreLib
+        plFile
+        string_theory
```

Comment:
`string_theory` already linked by `CoreLib`

Reply:
IMO, if a target directly consumes another target (in this case `plFileEncrypt` *directly* uses `string_theory`), then it ought to link against that target instead of hoping transitive linkage resolves it.

Reply:
For most of the spaghetti mess that is Plasma, I would agree...  But since string_theory is already prolific enough in Plasma (and including something like `hsStream.h`, `plFileSystem.h` or `hsWindows.h` will indirectly require it even if you don't explicitly include it separately), I think it's safe to assume CoreLib also brings in string_theory everywhere.

Reply:
OK. Just be sure - linking against what we directly consume is fine for the rest of the codebase?

Reply:
Yeah, I think that's fine for everything else.